### PR TITLE
goas: MMR for the zone transaction log

### PR DESCRIPTION
**Pull request by davidrusu**

### CODE REVIEW

File: goas/atomic_asset_transfer/common/src/mmr.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MMR {
+    pub roots: Vec<Root>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Root {
+    pub root: [u8; 32],
+    pub height: u64,
```
Comment (zeegomo) [inline goas/atomic_asset_transfer/common/src/mmr.rs:13]: nit: we could use a smaller type here

**Review by zeegomo — COMMENTED** [review]
Do we have benchmarks for this somewhere?

**Comment by davidrusu** [comment]
> Do we have benchmarks for this somewhere?

What do you want to benchmark? we measure the proofs themselves.

**Comment by zeegomo** [comment]
I meant the difference in proof cycles before/after

**Comment by davidrusu** [comment]
Sure thing! Here's the change in proving the Zone STF for different transactions:

| zone op | before total cycles | after total cycles | reduction |
|--|--|--|--|
| withdraw | 3145728 | 2228224 | 29% |
| deposit | 3145728 | 2162688 | 31% |

**Review by zeegomo — APPROVED** [review]

**Pull request merged**